includes/admin/shortners/class-rop-rebrandly-shortner.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * The file that defines this shortner specific functionality.
+ *
+ * @link       https://themeisle.com/
+ * @since      8.0.0
+ *
+ * @package    Rop
+ * @subpackage Rop/includes/admin/shortners
+ */
+
+/**
+ * Class Rop_Rebrandly_Shortner
+ *
+ * @since   8.0.0
+ * @link    https://themeisle.com/
+ */
+class Rop_Rebrandly_Shortner extends Rop_Url_Shortner_Abstract {
+
+	/**
+	 * Method to inject functionality into constructor.
+	 *
+	 * @since   8.0.0
+	 * @access  public
+	 * @return mixed
+	 */
+	public function init() {
+		$this->service_name = 'rebrand.ly';
+		$this->credentials  = array(
+			'key'  => '',
+		);
+	}
+
+
+	/**
+	 * Method to retrieve the shorten url from the API call.
+	 *
+	 * @since   8.0.0
+	 * @access  public
+	 * @param   string $url The url to shorten.
+	 * @return string
+	 */
+	public function shorten_url( $url ) {
+		$response = $this->callAPI(
+			'https://api.rebrandly.com/v1/links',
+			array( 'method' => 'json', 'json' => true ),
+			array( 'destination' => $url ),
```

Comment:
@contactashish13 Can you please add the param for custom domains which have been added to the rebrandly dashboard?

On the post format screen if rebrandly is chosen then show the user a text area where they can input their domain ID, we'll create a doc on how to get this (it can be found in the URL of the domains section on rebrandly). if no domain ID is used then it will just use the default rebrand.ly

The thing is rebrandly isn't just like the other shortners and people use it mostly for this purpose of rebranding their links to use their custom short domain.

---

Another good addition would be passing the title parameter to rebrandly as the WordPress post name, I believe you'd have to edit the get_short_url method accepted parameters a little to accomodate this addition.

The reason for this again is people use rebrandly to organize their links, without the tittle parameter rebrandly does a bad job at getting the destination page title and simply uses the short tag as the title:


![image](https://user-images.githubusercontent.com/10324144/43740878-8dab1b06-999a-11e8-9f4f-893e0c6d70ed.png)

Reply:
@UVLabs As per @selul's suggestion, I'm just using this as another shortener. If users want these features, I'm sure we can add these later. Suggestions @selul?

Reply:
yes, similar as we do add bitly api key, we can add the for rebrandly this domain id as a separate setting in the post format. makes sense @contactashish13 ?